utils/https.h

```diff
@@ -9,7 +9,9 @@
 #ifndef UTILS_HTTPS_H
 #define UTILS_HTTPS_H
 
+#include <stdio.h>
 #include "common/ta_errors.h"
+#include "http_parser.h"
```

Comment:
Is `http_parser.h` an implementation-oriented header? If so, remove the inclusion. Always minimize the header inclusions in public headers.